wrappers/services.go

```diff
@@ -900,22 +946,21 @@ type AddSnapServicesOptions struct {
 // AddSnapServices adds service units for the applications from the snap which
 // are services. The services do not get enabled or started.
 func AddSnapServices(s *snap.Info, opts *AddSnapServicesOptions, inter Interacter) error {
-	m := map[*snap.Info]*SnapServiceOptions{
-		s: {},
+	if opts == nil {
+		opts = &AddSnapServicesOptions{}
 	}
-	ensureOpts := &EnsureSnapServicesOptions{}
-	if opts != nil {
-		// set the per-snap service options
-		m[s].VitalityRank = opts.VitalityRank
-		m[s].QuotaGroup = opts.QuotaGroup
 
-		// copy the globally applicable opts from AddSnapServicesOptions to
-		// EnsureSnapServicesOptions, since those options override the per-snap opts
-		// we put in the map argument
-		ensureOpts.Preseeding = opts.Preseeding
-		ensureOpts.RequireMountedSnapdSnap = opts.RequireMountedSnapdSnap
+	m := map[*snap.Info]*SnapServiceOptions{
+		s: {
+			VitalityRank: opts.VitalityRank,
+			QuotaGroup:   opts.QuotaGroup,
+			Services:     MakeServiceQuotaMap(s.InstanceName(), s.Services(), opts.QuotaGroup),
```

Comment:
we need to do something else here, the main caller  in backend/link.go has code like this:

```
        var quotaGrp *quota.Group
        if linkCtx.ServiceOptions != nil {
                vitalityRank = linkCtx.ServiceOptions.VitalityRank
                quotaGrp = linkCtx.ServiceOptions.QuotaGroup
        }
        // add the daemons from the snap.yaml
        opts := &wrappers.AddSnapServicesOptions{
                VitalityRank:            vitalityRank,
                Preseeding:              b.preseed,
                RequireMountedSnapdSnap: linkCtx.RequireMountedSnapdSnap,
                QuotaGroup:              quotaGrp,
        }
```

it's basically copying bits of SnapServiceOptions, so we should do more of that in some form instead of doing this here

Reply:
Removed the usage of `AddSnapServices` in patch5.go, now `AddSnapServices` is simply just a test helper for services_test.go